test/role.test.js

```diff
@@ -489,48 +491,246 @@ describe('role model', function() {
     });
   });
 
-  it('resolves OWNER via "userId" property with no relation', function() {
-    var Album = app.registry.createModel('Album', {
-      name: String,
-      userId: Number,
-    });
-    app.model(Album, {dataSource: 'db'});
+  describe('$owner role resolver', function() {
+    var sender, receiver;
+    var users = [
+      {username: 'sender', email: 'sender@example.com', password: 'pass'},
+      {username: 'receiver', email: 'receiver@example.com', password: 'pass'},
+    ];
+
+    it('supports legacy behavior with no relation via property "userId"', function() {
+      var user;
+      var Album = app.registry.createModel('Album', {
+        name: String,
+        userId: Number,
+      });
+      app.model(Album, {dataSource: 'db'});
 
-    var user;
-    return User.create({email: 'test@example.com', password: 'pass'})
+      return User.create({email: 'test@example.com', password: 'pass'})
       .then(u => {
         user = u;
         return Album.create({name: 'Album 1', userId: user.id});
       })
       .then(album => {
         return Role.isInRole(Role.OWNER, {
-          principalType: ACL.USER, principalId: user.id,
-          model: Album, id: album.id,
+          principalType: ACL.USER,
+          principalId: user.id,
+          model: Album,
+          id: album.id,
         });
       })
       .then(isInRole => expect(isInRole).to.be.true());
-  });
-
-  it('resolves OWNER via "owner" property with no relation', function() {
-    var Album = app.registry.createModel('Album', {
-      name: String,
-      owner: Number,
     });
-    app.model(Album, {dataSource: 'db'});
 
-    var user;
-    return User.create({email: 'test@example.com', password: 'pass'})
+    it('supports legacy behavior with no relation via property "owner"', function() {
+      var user;
+      var Album = app.registry.createModel('Album', {
+        name: String,
+        owner: Number,
+      });
+      app.model(Album, {dataSource: 'db'});
+
+      return User.create({email: 'test@example.com', password: 'pass'})
       .then(u => {
         user = u;
         return Album.create({name: 'Album 1', owner: user.id});
       })
       .then(album => {
         return Role.isInRole(Role.OWNER, {
-          principalType: ACL.USER, principalId: user.id,
-          model: Album, id: album.id,
+          principalType: ACL.USER,
+          principalId: user.id,
+          model: Album,
+          id: album.id,
         });
       })
       .then(isInRole => expect(isInRole).to.be.true());
+    });
+
+    it('supports legacy behavior via a belongsTo relation', function() {
+      // passing no options will result calling
+      // the legacy $owner role resolver behavior
+      var Message = createModelWithOptions('ModelWithNoOptions');
```

Comment:
`createModelWithOptions` is too vague, it does not communicate the fact that the Model is set up with `belongsTo` relations via `senderId` and `receiverId`.

How about `createModelWithSenderReceiverRelations`? (Or even better, replace `create` with `given`  - see e.g. https://martinfowler.com/bliki/GivenWhenThen.html.)